### PR TITLE
Fix Django 1.9 test failures

### LINKED ISSUE #675: Django 1.9: Switch to Python's OrderedDict/import_module, Origin
I haven't done extensive testing but there are at least 3 problems when running the `compress` management command with Django 1.9a1:

---
```python
File "virtual/lib/python3.4/site-packages/compressor/management/commands/compress.py", line 13, in <module>
    from django.utils.datastructures import SortedDict
ImportError: cannot import name 'SortedDict'
```

https://code.djangoproject.com/wiki/SortedDict:

> SortedDict is deprecated as of Django 1.7 and will be removed in Django 1.9. Use ​collections.OrderedDict instead. Available in Python 2.7 and 3.1+.

---

```python
  File "virtual/lib/python3.4/site-packages/compressor/management/commands/compress.py", line 14, in <module>
    from django.utils.importlib import import_module
ImportError: No module named 'django.utils.importlib'
```

import_module can be replaced by [Python's importlib.import_module](https://docs.python.org/2/library/importlib.html)

---

At this point it says "No templates found" because of a crash on line 140:
```python
  File "virtual/lib/python3.4/site-packages/compressor/management/commands/compress.py", line 139, in compress
    paths.update(list(get_template_sources('')))
TypeError: unhashable type: 'Origin'
```

I worked around this by changing that line to 
```python
paths.update([x.name for x in get_template_sources('')])
```

### LINKED ISSUE #676: Django 1.9 compatibility blocked by `overextends` dependency.
#670 introduced a tests-only dependency on django-overextends. This is currently blocking 1.9 compatibility work because `overextends` is not 1.9 compatible itself. 

```
django-compressor/.tox/py34-1.9.x/lib/python3.4/site-packages/overextends/models.py", line 5, in <module>
    from django.template.base import add_to_builtins
ImportError: cannot import name 'add_to_builtins'
```

The issue is here: stephenmcd/django-overextends#19

### PR DESCRIPTION
**Pull request by karyon**
fixes #675, fixes #676

closes #697 which is included
closes #695 which is kinda included as well

@diox, could you have a look at the remaining test failures? one is that the overextends tag could not be found, the other that some hashes are not in the manifest. you have any idea about either of these?

### CODE REVIEW

File: compressor/tests/test_offline.py [hunk 1]
```diff
@@ -589,31 +590,6 @@ def _test_offline(self, engine):
...
-
-        # Could have used the env.add_extension method, but it's only available
-        # in Jinja2 v2.5
-        new_env = jinja2.Environment(extensions=[CompressorExtension])
-        env.extensions.update(new_env.extensions)
-
-        return env
-
-
 # Jingo does not work when using Python 3.2 due to the use of Unicode string
```
Comment (diox) [inline compressor/tests/test_offline.py]: You can remove Jingo stuff too.

**Comment by diox** [comment]
Missing hash could be an error in the compress step that was hidden somehow. If that happens, manifest would be empty and raise that error. There are other possibilities, but that scenario is the easiest to look for / fix, so let's hope it's what's happening here :)

Regarding overextends, I have no idea. I'm tempted to skip those tests for now (especially as it doesn't have an official 1.9 release yet)

**Comment by diox** [comment]
See also: #692

**Comment by karyon** [comment]
there in one more problem: the old template settings are removed in django 1.10 ( that's the one remaining deprecation warning). i attempted to update all settings in the tests to the new TEMPLATES setting, but there doesn't seem to be a nice way to override some elements in the TEMPLATES setting, and that is done all over the place in compressor's tests...

i simply don't know how i should do it. as last resort i'd just leave it as it is and live with the deprecation warning, but in 8 months (well, preferably earlier)  this needs to be done anyways for the next django release...

**Comment by karyon** [comment]
see the last commit for a glimpse on the necessary changes. it gets even worse because there are tests defining `additional_setting` whose handling will be even uglier i think.

**Comment by diox** [comment]
Yeah, I actually introduced that `additional_settings` property to make the settings overrides in those tests more sane but when I wrote it I realized the horror it was going to be later when switching to `TEMPLATES` ... I don't have a good suggestion as to how to handle that mess yet.

**Comment by diox** [comment]
I just merged #692 to get Python 3.5 tests working so you'll have to rebase. To get Django 1.9 with Python 3.5 you'll want to add to .travis.yml, in the matrix -> includes directive:
```
    - python: 3.5
      env:
        - TOXENV=py35-1.9.X
```

**Comment by karyon** [comment]
I think that's it, please review again.

the test failures were all due to the incompatibility between django 1.9, django-overextends and the old templates settings format. i did not update the settings, so i disabled the overextends-tests for django 1.9. i think as soon as the tests are updated to use the new settings format it should work (and the last remaining warning will be fixed). i'd open a new issue for that.

**Comment by diox** [comment]
This is awesome work, thanks.

**Comment by frgtn** [comment]
Thank you! :+1:

**Pull request merged**